components/wordpress_org/wordpress_org.app.mjs

```diff
@@ -1,11 +1,203 @@
+      description: "Display name for the user",
+      optional: true,
+    },
+    firstName: {
+      type: "string",
+      label: "First Name",
+      description: "First name for the user",
+      optional: true,
+    },
+    lastName: {
+      type: "string",
+      label: "Last Name",
+      description: "Last name for the user",
+      optional: true,
+    },
+    email: {
+      type: "string",
+      label: "Email",
+      description: "  The email address for the user",
+    },
+    url: {
+      type: "string",
+      label: "URL",
+      description: "URL of the user",
+      optional: true,
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "Description of the user",
+      optional: true,
+    },
+    roles: {
+      type: "string[]",
+      label: "Roles",
+      description: "Roles assigned to the user",
+      optional: true,
+    },
+    password: {
+      type: "string",
+      label: "Password",
+      description: "  Password for the user",
+    },
+    author: {
+      type: "string",
+      label: "Author",
+      description: "Limit result set to posts assigned to a specific author",
+      optional: true,
+      async options({ page }) {
+        const authors = await this.getAuthors(page + 1);
+        return authors.map((author) => ({
+          label: author?.name,
+          value: author?.id,
+        }));
+      },
+    },
+    status: {
+      type: "string",
+      label: "Status",
+      description: "Status of the post",
+      optional: true,
+      async options() {
+        const statuses = await this.listStatuses();
+        return Object.keys(statuses).map((key) => ({
+          label: statuses[key]?.name,
+          value: statuses[key]?.slug,
+        }));
+      },
+    },
+    categories: {
+      type: "string[]",
+      label: "Categories",
+      description: "Categories assigned to the post",
+      optional: true,
+      async options({ page }) {
+        const categories = await this.listCategories(page + 1);
+        return categories.map((category) => ({
+          label: category?.name,
+          value: category?.id,
+        }));
+      },
+    },
+    post: {
+      type: "string",
+      label: "Post",
+      description: "The post to update",
+      async options({ page }) {
+        const posts = await this.listPosts(page + 1);
+        return posts.map((post) => ({
+          label: post?.title?.rendered,
+          value: post?.id,
+        }));
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async getClient() {
+      const {
+        url,
+        username,
+        application_password: applicationPassword,
+      } = this.$auth;
+      return WPAPI.discover(`https://${url}`).then(function(site) {
+        return site.auth({
+          username,
+          password: applicationPassword,
+        });
+      });
```

Comment:
The standard is to always use `async/await` instead of promises as much as possible
```suggestion
      const site = await WPAPI.discover(`https://${url}`);
      return site.auth({
        username,
        password: applicationPassword,
      });
```